include/pressio/ops/eigen/ops_fill.hpp

```diff
@@ -55,14 +55,22 @@ namespace pressio{ namespace ops{
   native Eigen ops which use expressions, so value has to
   be convertible to the the scalar type of the object */
 template <typename T, class ScalarType>
+#ifdef PRESSIO_ENABLE_CXX20
+requires (::pressio::is_native_container_eigen<T>::value
+       || ::pressio::is_expression_acting_on_eigen<T>::value)
+       && std::convertible_to<ScalarType, typename ::pressio::Traits<T>::scalar_type>
+void
+#else
 ::pressio::mpl::enable_if_t<
     (::pressio::is_native_container_eigen<T>::value
   || ::pressio::is_expression_acting_on_eigen<T>::value)
   && std::is_convertible<ScalarType, typename ::pressio::Traits<T>::scalar_type>::value
   >
+#endif
 fill(T & o, const ScalarType & value)
 {
-  impl::get_native(o).setConstant(value);
+  const auto v = static_cast<typename ::pressio::Traits<T>::scalar_type>(value);
```

Comment:
i think this is fine, but i am wondering if we can just simplify to: 
```
const typename ::pressio::Traits<T>::scalar_type v(value);
```

Reply:
this would actually be interesting to see on godbolt if doing a static cast and calling the copy constructor generates equilvant assembly code to just calling the copy constructor via implicit conversion . can you try?

Reply:
Thanks - it's a good point. Casts simplified with 2f58cf2a80caa7f8e36d89d0daa3f984b589188d. 

Here's _clang-15_ output for [`double`](https://godbolt.org/#z:OYLghAFBqd5TKALEBjA9gEwKYFFMCWALugE4A0BIEAZgQDbYB2AhgLbYgDkAjF%2BTXRMiAZVQtGIHgBYBQogFUAztgAKAD24AGfgCsp5eiyahUAUgBMAIUtXyKxqiIEh1ZpgDC6egFc2TA3cAGQImbAA5PwAjbFIQAE5yAAd0JWIXJi9ffwMUtOchELDIthi4xIdsJwyRIhZSIiy/AJ57bEcCplr6oiKI6NiE%2BzqGppzWpRHe0P7SwfiASnt0H1JUTi5MFajGAGoANQBBIN2zAGYAEV2eM/ObLUOzB9CiXbYWUIgF04ezAHZ7oddsDdhgmJNdlsfDtsLsAG4QI5BBZ3J5AkFI06XeGoh4g3akbBEVZMA7HLG4XYAVixADFrrsQLstLjHn8LmiuEt6NwqfwAlwdOR0NwPLZbLslCs1rDLGc%2BOQiNouUsANYgKlaQzcaT8NgarUCoUirj8JQgLVKwVc8hwWAoDA4fDEMiUah0RisDjcBWCYRiCScGRyYTKNSaa3kfQWQzGUzi6xtDoZNxMTzeZqBNN9EplIZ5dJCMYtZKpQtMHMDcpJqqdbqjDPjGvVIT16bFKtDSY9YsGbsNStzcpLKWrdbcKEwsknc5XG6stEvN4fJhfH5swH4sEQlg%2BEjwrFXbvOVAAfXEk3OHknjHOuERxxRZ03GPJs5xz7R%2BMJxNIpMxd7UnSDJMiyn6/OynLcry/LKsKooJlYkrSuspwWGcFj8FaOgLEsSDYCwOBxF82pcLq5D6pq5BGvwJpmhairKrh5DqtIAAcAB0Wh/BYvFsdI3E8EJWhnKRZywZGdGMdaSx2sgaDoGwSQMLEboQBgSkqXEqBGCYPCalqOBwgQ6z7AQ2AAO4APJJMwPr8B6RCxOaEBRHBUShPUACe9nkBpHDCFZTD0D5kY4O8JiSGFBCEtUcLYOakbYOoVR7hsCovO0cH0AQUSkN5Xg4HBRCkAQ%2Bq8DaNBGMAShmZZNl2RVIaiOIkjBn6igqBocH6K0unxtYtiGLl5qQEs6BJJ0iUALRWeJzadKm6bZAEABsWrBDMuaDDwWoFp0vbraW%2BQZIOea7QtNRTIdWqVC2XRTGdO23ddjZrS9PRPXEu0jihQbQVwfLUXBJq7OobGrdNq3SKC/XXFSXFcbsEDOiQpBofKCxYUxtoOopymMBQVDqfjWloP1%2BlaIZ2DGaZ5nWbZAq%2BgwTmkC5bmRh5rCkKFCr%2BcwRBBSFcHhXGUVCoQsXOPFiVCslqVOb5mU8pGOV5QVWAbEKJVlfZSxVSwNV1QzjW%2BvIAZtbIHVht1kbRrGJhoIhw1RKNJETVN3CzfNd2LRA7i9q0m0dkOuRlgdb1hydhRbZ2EztLWV09pH8fJq2j2x6HEyvStfYZyH52/WOnAWADQM0fBXBgxDUMwzpcbw4jWjI6jZAY6X2MybjICEDQNBqR1FtBlb8g2xGQr0Eg5p9VPvc0EQXm2QxpBTyAMYr0oc8L0vWhlxJxrcBcBB9wc9OxNXkPQ7DDf6U38JKBftfXyYjdaFxnc4XhBFEdQaoajwHF4gWD%2BCA/SQC2JsUSCrci%2BpdqGhBtweilomIA0wsDSSiDpKf3IPFNmKZpBAA%3D%3D) (fundamental type representative) and for [`std::complex<double>`](https://godbolt.org/#z:OYLghAFBqd5TKALEBjA9gEwKYFFMCWALugE4A0BIEAZgQDbYB2AhgLbYgDkAjF%2BTXRMiAZVQtGIHgBYBQogFUAztgAKAD24AGfgCsp5eiyahUAUgBMAIUtXyKxqiIEh1ZpgDC6egFc2TEABmLXJ3ABkCJmwAOT8AI2xSIPIAB3QlYhcmL19/IJC0jOchCKjYtgSkwPtsR2KmESIWUiIcvwDgmrqsxuaiUpj4xOSlJpa2vM7RvoHyyqCASnt0H1JUTi5LQMjUXxwAajNAjww2FMZ1I9wzLQBBG9vRzBA0dDOLo49MFbjGK/2AGq3MKHQIAEX2PEC%2BwAtPsLAQjjY7g9IkR9mwWJEIAtDgB2ZG3fbE/YYJijfZPF6nc7YS7Hb4%2BX7Yf4ANwgQLCCyRDxJ%2B1I2CIqyY%2B1ZoLBRwhVNe7zpn0ZzKuEAArLijgAxSH7ED7LQ8lF4iV3LhLejcZX8AJcHTkdDcDy2WyUlZrbCHCyBPjkIjaE1LADWIGVITNXGk/DYQZCVptdq4/CUIBCPutJvIcFgKAwOHwxDIlGodEYrA43C9gmEYgknBkcmEyjUmlT5H0FkMxlMjusXWwTiybiYnm87SCPFCg9mQySeNS6UyQgmHTHhXnTEnFWGM4cvfqvXGw8mY%2B3faEe/6kUGG%2Bn9jGrQPS5vMwvc03SyULvW3C2Oz2bs%2BNI%2BQJrhRO5pQAuUGR%2BP4gMBYFxUhaE4QRfV7juNEMSxJgcUOA1CT5MkKRYHwSFFeDpmcVAAH1xFGT4wLeWl6S%2BKCWSApVOW5QI8JJAUhVIEUxUlSVKSIZ4ZUY%2BUWKVVVQU1HhtV1FCzENB401DC1yBjfg4wdawnXfVZ1ndQILH4FMdAWJYkGwFgcCSHFDG4cNyEjYNNN9W1uATJNvV9SzyEDYIADoLGkHgADY8QsHgtAATjxEzotkUNAktDy4zMvz00QCAswYhhEgLCAaQKpJdg7HhgxCHBWQIdYAQIbAAHcAHkUmYMt%2BCLIhEkTCA4g8uJImaABPTryFODhhBaph6DG5scExExJAWggBT7VlsETZs6V7YiNi9NFag8%2BgCDiUhRq8HAPKIUgCEjXg0xoIxgCUBrmrajrHrrURxEkWsK0UFQNA81t2xMNAuzsU64kTSAlnQFJ6m2mEWtSnsTwCCB3EXAxwmfKcDBXepceXOd6nXeYj1qHcelvUmMd3W9KeGI96fvAxphaFmkh4N8P04CxTXNNLmzjfZ1AADnCmFwukUkjBMSFlSCrRVf2CBcxIUhjKFzLUyWXLxNKoqSsYMrFeASqtGq7BavqxrWvaq1ywYHrSD6gbmyG1hSHmr1JuYIgZrmjzFo7FabUIdbnE27abV21B9vGo7QxtGGLr9q6NhtW77s6pZnpYV73qdr7y3kKt/tkQGGxB5t9DHS3Ib07sYbhhzEeR7hUfR496gHIdcgCMd8bKQmyaKLIGeJrIec5mnMbPBn%2B7pp9x6vTn2eHrf18vKn%2BcMmtha4DStM8rgJel2X5fKpXKtV9XNcIbXjL5/WLPTFBCBoGgisBquNYa7yDrk2dOSBExNwgT/GgRARrtR8qQCBIA2xIKUDAuBCCtAnzPulbgYICC/0BI7RIV8ZZywVh2ZWj8tCiiUGQm%2BlD74qzVrnPyVkbJ2WoAGEA4VYpBTxMqSWJk8SS1iloCwkVJZNychGKQNt3Jiy8vYFBvkDYn1Moo2MyjzJ%2BnIJtD2/ZpBAA) (complex type representative) - no difference in generated code.

Could be hard to obtain the difference as `static_cast<T>(e)` - by [the rules](https://en.cppreference.com/w/cpp/language/static_cast) - will usually resolve to `T temp(e)` here in our context (with no pointers/references nor `const`/`volatile` involved), providing little added value.